src/agent/src/linux_abi.rs

```diff
@@ -15,9 +17,32 @@ pub const SYSFS_PCI_BUS_RESCAN_FILE: &str = "/sys/bus/pci/rescan";
     target_arch = "x86_64",
     target_arch = "x86"
 ))]
-pub const PCI_ROOT_BUS_PATH: &str = "/devices/pci0000:00";
+pub fn create_pci_root_bus_path() -> String {
+    String::from("/devices/pci0000:00")
+}
+
 #[cfg(target_arch = "aarch64")]
-pub const PCI_ROOT_BUS_PATH: &str = "/devices/platform/4010000000.pcie/pci0000:00";
+pub fn create_pci_root_bus_path() -> String {
+    let mut sysfs_dir = String::from(SYSFS_DIR);
+
+    let mut start_root_bus_path = String::from("/devices/platform/");
+    let end_root_bus_path = String::from("/pci0000:00");
+
+    sysfs_dir.push_str(&start_root_bus_path);
+    for entry in fs::read_dir(sysfs_dir).unwrap() {
```

Comment:
You'd better deal with the error case instead of unwrap it directly, otherwise, it would panic the program, this isn't what we want.

Reply:
thanks @lifupan -, does that mean canceling all the "unwrap()" and use match statements instead? just like: 

let entries = match fs::read_dir(sysfs_dir) {
        Ok(e) => e,
        Err(_) => return ret,
    };
    for entry in entries {
        let pathname = match entry {
        Ok(p) => p.path(),
        Err(_) => return ret,
    };
        let pathname = match pathname.file_name() {
        Some(p) => p.to_str(),
        None => return ret,
    };
        let pathname = match pathname {
        Some(p) => p,
        None => return ret,
        };
        let dir_name = String::from(pathname);

Reply:
Yes, if you can make sure the unwrap is safe, you can unwrap it directly, otherwise, you should catch the Error/None case.

Reply:
thanks, I have addressed it.